### DISCUSSION TITLE
Edit a Select() prior to sending a Modal

### DISCUSSION THREAD

**Discussion by ethanlchristensen** [Rapptz/discord.py#9289] [Q&A] [answered]
Is it possible to disable a Select() that is in a view, prior to sending a modal? That is, the View which contains the Select() UI has a call back that sends the Modal. 

In the code below, the user will select an option, and then this callback function is called. Is it possible to disable this selector? prior to sending this modal?

Now, if I update the view, and pass it into the modal, I can disable that selector that way, but this only get's disabled after the Modal has been sent and submitted. 
```py
async def callback(interaction):
            weather_select.disabled = True
            weather_select.placeholder = "thinking . . ."
            typE = weather_select.values[0]
            modal = WeatherModal(
                typE=typE,
                prompt="City: ",
                title="Enter a City"
            )
            await interaction.response.send_modal(modal)
```

**Comment by ika2kki** [comment, marked as answer]
sorry for the late response !

[Interaction.edit_original_response](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.Interaction.edit_original_response) can be called to edit the message, since `response.send_modal()` causes the original message to point to the message with the component, for example:
```py
await interaction.response.send_modal(modal)
weather_select.disabled = True
await interaction.edit_original_response(view=weather_select.view)
```

**Reply by ethanlchristensen** [reply]
I suppose a better phrased question would have been, can you disable the Select while the user is editing the modal? Would this accomplish that? I will try this later today. The select menu it what will prompt the modal. The user makes a selection, then the modal is sent. I am assuming then that I could edit the original response to disable to select, then send the modal?

**Reply by ethanlchristensen** [reply]
That is, while the user is filling out the modal, the select can no longer be selected

**Reply by ika2kki** [reply]
> That is, while the user is filling out the modal, the select can no longer be selected

Ya, this is what this does. It sends the modal + disable the select.
